src/kixi/comms/components/kafka.clj

```diff
@@ -108,51 +108,61 @@
-                                :enable.auto.commit      true
-                                :auto.commit.interval.ms timeout}
-            listener            (callbacks/consumer-rebalance-listener
-                                 (fn [tps]
-                                   (info "topic partitions assigned:" tps))
-                                 (fn [tps]
-                                   (info "topic partitions revoked:" tps)))
-            co                 (cd/make-default-consumer-options
-                                {:rebalance-listener-callback listener})
-            consumer           (consumer/make-consumer
-                                cc
-                                key-deserializer
-                                value-deserializer)
-            running?            (atom true)]
-        (cp/subscribe-to-partitions! consumer (vals topics))
-        (async/go (async/<! kill-chan) (reset! running? false))
-        (loop []
-          (let [cr (into [] (cp/poll! consumer {:poll-timeout-ms timeout}))]
-            (loop [cr' cr]
-              (when-let [process (first cr')]
-                (if @running?
-                  (do
-                    (async/put! out-chan ((comp transit->clj :value) process))
-                    (recur (rest cr')))
-                  (do
-                    (cp/commit-offsets-sync! consumer {(select-keys process [:topic :partition])
-                                                       {:offset (:offset process)
-                                                        :metadata (str "Consumer stopping - "(java.util.Date.))}}))))))
-          (if @running?
-            (recur)
-            (do
-              (cp/clear-subscriptions! consumer)
-              (.close consumer)))))
-      (catch Exception e 
-        (error e "Consumer exception")))))
+  [kill-chan group-id {:keys [commands events] :as topics} broker-list 
+   msg-type event version handler]
+  (let [timeout 100
+        cc {:bootstrap.servers       broker-list
+            :group.id                group-id
+            :auto.offset.reset       :earliest
+            :enable.auto.commit      false
+            :auto.commit.interval.ms timeout}
+        listener (callbacks/consumer-rebalance-listener
+                  (fn [tps]
+                    (info "topic partitions assigned:" tps))
+                  (fn [tps]
+                    (info "topic partitions revoked:" tps)))
+        co (cd/make-default-consumer-options
+            {:rebalance-listener-callback listener})
+        consumer (consumer/make-consumer
+                  cc
+                  (deserializers/string-deserializer)
+                  (deserializers/string-deserializer))
+        _ (cp/subscribe-to-partitions! consumer (vals topics))]
+    (async/go-loop []
+      (let [[val port] (async/alts! [(async/thread 
+                                       (cp/poll! consumer {:poll-timeout-ms timeout}))
```

Comment:
Surprised this works. http://www.kimkha.com/2016/07/kafkaconsumer-is-not-safe-for-multi.html